File: aes/php/aes_ecb.php
<?php

require_once "Crypt.php";

//$key = 'pwd';
//$des = new Crypt($key, MCRYPT_DES, MCRYPT_MODE_CBC);//DES
//$des = new Crypt($key);//AES，默认是MCRYPT_RIJNDAEL_128+MCRYPT_MODE_ECB
//echo $ret = $des->encrypt("123456").PHP_EOL;//加密字符串，结果默认已经base64了
//echo $ret = $des->decrypt($ret);//解密结果
//echo PHP_EOL.'--------------'.PHP_EOL;

$key = '1234567812345678';
$des = new Crypt($key, MCRYPT_RIJNDAEL_128, MCRYPT_MODE_ECB, '1234567812345678');//DES
echo $ret = $des->encrypt("123456").PHP_EOL;// mdSm0RmB+xAKrTah3DG31A==
echo $ret = $des->decrypt($ret);//解密结果  123456
echo PHP_EOL.'--------------'.PHP_EOL;

File: aes/python/aes_ecb.py
# -*- coding=utf-8-*-
from Crypto.Cipher import AES
import os
from Crypto import Random
import base64

"""
aes加密算法
padding : PKCS7
"""

class AESUtil:

    __BLOCK_SIZE_16 = BLOCK_SIZE_16 = AES.block_size

    @staticmethod
    def encryt(str, key, iv):
        cipher = AES.new(key, AES.MODE_ECB,iv)
        x = AESUtil.__BLOCK_SIZE_16 - (len(str) % AESUtil.__BLOCK_SIZE_16)
        if x != 0:
            str = str + chr(x)*x
        msg = cipher.encrypt(str)
        # msg = base64.urlsafe_b64encode(msg).replace('=', '')
        msg = base64.b64encode(msg)
        return msg

    @staticmethod
    def decrypt(enStr, key, iv):
        cipher = AES.new(key, AES.MODE_ECB, iv)
        # enStr += (len(enStr) % 4)*"="
        # decryptByts = base64.urlsafe_b64decode(enStr)
        decryptByts = base64.b64decode(enStr)
        msg = cipher.decrypt(decryptByts)
        paddingLen = ord(msg[len(msg)-1])
        return msg[0:-paddingLen]

if __name__ == "__main__":
    key = "1234567812345678"
    iv = "1234567812345678"
    res = AESUtil.encryt("123456", key, iv)
    print res # mdSm0RmB+xAKrTah3DG31A==
    print AESUtil.decrypt(res, key, iv) # 123456
